src/Control/Carrier.hs

```diff
@@ -1,12 +1,22 @@
+{-# LANGUAGE ConstraintKinds #-}
 module Control.Carrier
 ( -- * Re-exports
   module Control.Carrier.Class
+, module Control.Carrier.Pure
 , module Control.Effect.Class
-, module Control.Effect.Sum
-, run
+, (:+:)(..)
+, Has
+, send
 ) where
 
 import Control.Carrier.Class
 import Control.Carrier.Pure
 import Control.Effect.Class
 import Control.Effect.Sum
+
+type Has eff sig m = (Inject eff sig, Carrier sig m)
+
+-- | Construct a request for an effect to be interpreted by some handler later on.
+send :: Has eff sig m => eff m a -> m a
+send = eff . inj
```

Comment:
I moved this back out here because `Has` definitely doesn’t belong in `Control.Effect.Sum`.